### PR TITLE
Correct spawning of humans in jobs.

### LINKED ISSUE #16821: Humans unable to spawn as captain
Suppose a new player connects to a server for the first time, creates their character, and joins.

If the server config allows joining only as strict humans, players may find themselves **unable to join as heads.**

```
#JOIN_WITH_MUTANT_RACE
#JOIN_WITH_MUTANT_HUMANS
```

I traced the bug to roughly this line -- 
https://github.com/tgstation/-tg-station/blob/master/code/modules/client/preferences.dm#L489

The condition for not letting someone join as a job checks against their pref race, _which may well be disabled in the server config,_ rather than the effective race that they _will spawn as._

So we need to put a fix into either
1. `preferences.dm` , making that if() line that much longer
2. `code/modules/mob/living/carbon/human/species_types.dm`, where the qualification/disqualification gets thrown out if the player _will inevitably spawn as human._

I favor updating `/datum/species/proc/qualifies_for_rank()` to return the fluff reason for rejection (NON-HUMAN/MUTANT)

I really want to turn qualifies_for_rank inside out, so it returns 0 if the rank IS allowed, but I don't have a good name for the proc . Advantage: return `null` when the species qualifies for rank, and `"MUTANT"` or `"NON-HUMAN"` as appropriate directly from the species proc.

### PR DESCRIPTION
**Pull request by octareenroon91**
if() statements were checking prefs instead of looking at what a player would expect to spawn as.

Fixes #16821

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -114,10 +114,10 @@
 
 
 //Please override this locally if you want to define when what species qualifies for what rank if human authority is enforced.
-/datum/species/proc/qualifies_for_rank(rank, list/features)
+/datum/species/proc/does_not_qualify_for_rank(rank, list/features)
 	if(rank in command_positions)
-		return 0
-	return 1
+		return "NON-HUMAN"
+	return null
```
Comment (duncathan) [inline code/modules/mob/living/carbon/human/species.dm]: probably better to return "" for consistency but it's not a big deal

Comment (Giacom) [inline code/modules/mob/living/carbon/human/species.dm]: I think an empty string would return false.

I also think this proc should be renamed, the name indicates that it returns a true/false statement.

**Comment by octareenroon91** [comment]
I am open to alternative proc names. It's one of the weakest parts of this PR.

Also, I'm pretty sure "" is allowed as a 'false'. I picked null thinking it was more obvious, but blank string is probably more correct.

**Comment by Giacom** [comment]
I would rename it to

```C++
/proc/get_disqualification_reason(rank, list/features)
```

I would then make another proc, just for testing whether they are qualified.

```C++
/proc/is_qualified_for_rank()
    return get_disqualification_reason(rank, list/features) == null
```

And use that purely for if() checks.

**Comment by octareenroon91** [comment]
I can see how this might be too "clever". It still feels funny to have two procs that will mostly be used in tandem, in place of one.

File: code/modules/mob/living/carbon/human/species.dm (hunk 1 above)
Comment (duncathan) [inline code/modules/mob/living/carbon/human/species.dm]: Yes an empty string is considered false

**Comment by Cheridan** [comment]
What's the status on this @octareenroon91

**Comment by octareenroon91** [comment]
Incorporating changes today

**Comment by octareenroon91** [comment]
Code is sound, but it doesn't fix the bug.

**Pull request closed without merging**